geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloFeatureWriter.scala

```diff
@@ -208,28 +222,30 @@ class ModifyAccumuloFeatureWriter(featureType: SimpleFeatureType,
   /* only write if non null and it hasn't changed...*/
   /* original should be null only when reader runs out */
   override def write() =
-    if(!live.equals(original)) {  // This depends on having the same SimpleFeature concrete class
-      if(original != null) remove()
+    // comparison of feature ID and attributes - doesn't consider concrete class used
+    if (!ScalaSimpleFeature.equalIdAndAttributes(live, original)) {
```

Comment:
ScalaSimpleFeatures aren't tied directly to kryo... but the method here compares 2 SimpleFeatures of any impl. I put it in ScalaSimpleFeature as a decent place to live, but maybe somewhere else would be better.